### ISSUE TITLE
Init.d script for Varnish 4.1.2-1~precise is broken

### ISSUE THREAD

**Issue by Rylon** [varnishcache/varnish-cache#1903]
Hi,

Not sure if this is the correct place to raise this issue, but I noticed when upgrading to Varnish `4.1.2-1~precise` that the provided init.d script contains a small error around the pidfile.

In the previous init.d script, the top section looked like this:

```
NAME=varnishd
DESC="HTTP accelerator"
PATH=/sbin:/bin:/usr/sbin:/usr/bin
DAEMON=/usr/sbin/varnishd
PIDFILE=/run/$NAME.pid
```

But in this new version, `$NAME` was partially changed to `$SERVICE`:

```
SERVICE="$(basename ${0})"
DESC="HTTP accelerator"
PATH=/sbin:/bin:/usr/sbin:/usr/bin
DAEMON=/usr/sbin/varnishd
PIDFILE=/run/$NAME.pid
```

This causes issues with things such as Monit, or upgrades, because the script is using the wrong path for the pidfile.

## Expected Behavior

The pidfile should exist at `/var/run/varnish.pid`.

## Current Behavior

The pidfile exists at `/var/run/.pid`.

## Possible Solution

Change `PIDFILE=/run/$NAME.pid` to `PIDFILE=/run/$SERVICE.pid`

## Context

We're rolling out Varnish via config management, and it caused our automation to fail. I believe this is because the postinstall script tries to restart Varnish using this new init.d script, and because it's looking in the wrong location for the existing PID, it fails to realise Varnish is already running, and tries to start it again. This fails because the port is already in use, so the package installation fails to complete.

## Your Environment

* Version used: 4.1.2-1~precise
* Operating System and version: Ubuntu 14.04.1 LTS

I'm not exactly sure why we're installing the Precise version on Trusty, but I assume the init.d script is the same on both?

Thanks

**Comment by lkarsten** [comment]
Hi, and thanks for the report.

Packaging bugs should be reported on https://github.com/varnishcache/pkg-varnish-cache/ , but I think isn't written down anywhere.

I've commited your fix here: https://github.com/varnishcache/pkg-varnish-cache/commit/f74bc84a925b42c4902c93b3f231fecc93379527

**Comment by lkarsten** [comment]
Rebuild packages are in jenkins now, will be on repo.v-c.o after some minor deployments problems are cleaned out. (internal reference: ZD 13290)

**Comment by Rylon** [comment]
@lkarsten Thanks! Sorry for raising in the wrong place initially, next time I'll search harder for it, and send a pull request 👍 

Do you know roughly when I can expect the packages to appear on http://repo.varnish-cache.org/ubuntu/pool/varnish-4.1/v/varnish/ ?

**Comment by lkarsten** [comment]
Hi. Updated packages are available in the repository now under version 4.1.2-2.

Please let me know if you find any more issues.

Closing this as completed.

**Comment by Rylon** [comment]
Thank you! :)
